pkg/daemon/daemon.go

```diff
@@ -745,8 +745,15 @@ func (dn *Daemon) runPreflightConfigDriftCheck() error {
 	}
 
 	currentOnDisk, err := dn.getCurrentConfigOnDisk()
-	if err != nil {
-		return fmt.Errorf("could not get on-disk config: %w", err)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
+	if currentOnDisk == nil {
+		currentOnDisk, err = dn.substituteMissingODC()
+		if err != nil {
+			return fmt.Errorf("could not get the state: %w", err)
```

Comment:
This will duplicate with the error message inside of the function. I think you can just return err directly here.